components/raftstore/src/store/fsm/apply.rs

```diff
@@ -4112,9 +4121,15 @@ where
             normal.delegate.id() == 1003,
             |_| { HandleResult::KeepProcessing }
         );
-        while self.msg_buf.len() < self.messages_per_tick {
+        let mut total_size = 0;
+        while self.msg_buf.len() < self.messages_per_tick
+            && total_size < self.messages_size_per_tick
+        {
             match normal.receiver.try_recv() {
-                Ok(msg) => self.msg_buf.push(msg),
+                Ok(msg) => {
+                    total_size += msg.entries_size();
```

Comment:
Why not check batch size directly by moving the handle task logic into this loop

Reply:
Sorry, I don't get your point. Are there any benifit to do so?

Reply:
msg's size doesn't mean the real size written to db, but batch size does